101-vm-ubuntu-DSVM-GPU-or-CPU/azuredeploy.json

```diff
@@ -0,0 +1,255 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "adminUsername": {
+            "type": "string",
+            "metadata": {
+                "description": "Username for Administrator Account"
+            }
+        },
+        "adminPassword": {
+            "type": "secureString",
+            "metadata": {
+                "description": "Password for for Administrator Account"
+            }
+        },
+        "vmName": {
+            "type": "string",
+            "defaultValue": "vmName",
+            "metadata": {
+                "description": "The name of you Virtual Machine."
+            }
+        },
+        "location": {
+            "type": "string",  
+            "defaultValue": "[resourceGroup().location]",  
+            "metadata": {  
+                    "description": "Location for all resources."  
+            }  
+        },
+        "Cpu-Gpu": {
+            "type": "string",
+            "defaultValue": "cpu",
+            "allowedValues": [
+                "cpu",
+                "gpu"
+            ],
+            "metadata": {
+                "description": "Choose between CPU or GPU processing"
+            }
+        },
+        "virtualNetworkName": {
+            "type": "string",
+            "defaultValue": "vNet",
+            "metadata": {
+                "description": "Name of the VNET"
+            }
+        },
+        "subnetName": {
+            "type": "string",
+            "defaultValue": "subnet",
+            "metadata": {
+                 "description": "Name of the subnet in the virtual network"
+            }
+        },
+        "networkSecurityGroupName": {
+            "type": "string",
+            "defaultValue": "SecGroupNet",
+            "metadata": {
+                "description": "Name of the Network Security Group"
+            }
+        }
+    },
+    "variables": {
+        "networkInterfaceName": "[concat(parameters('vmName'),'NetInt')]",
+        "virtualMachineName": "[parameters('vmName')]",
+        "publicIpAddressName": "[concat(parameters('vmName'), 'PublicIP' )]",
+        "subnetRef": "[resourceId('Microsoft.Network/virtualNetworks/subnets', parameters('virtualNetworkName'), parameters('subnetName'))]",
+        "nsgId": "[resourceId('Microsoft.Network/networkSecurityGroups', parameters('networkSecurityGroupName'))]",
+        "osDiskType": "Standard_LRS",
+        "storageAccountName": "[concat('storage', uniqueString(resourceGroup().id))]",
+        "storageAccountType": "Standard_LRS",
+        "storageAccountKind": "Storage",
+        "vmSize": {
```

Comment:
let's make this an explicit user choice and expand the list - otherwise we're going to end up blocking or failing..

Reply:
Do you mean the naming of the storageAccount properties?